src/pages/StakeTri/index.tsx

```diff
@@ -0,0 +1,294 @@
+  // 2)   Check for sufficient funds
+  // 3)   Approve | Approve Pending
+  // 4)   Stake/Unstake
+  function buttonRenderer() {
+    // If no wallet, show connect wallet
+    if (account == null) {
+      return (
+        <ButtonLight onClick={toggleWalletModal}>
+          Connect Wallet
+        </ButtonLight>
+      );
+    }
+
+    // If input does not have value
+    if (parsedAmount?.greaterThan(JSBI.BigInt(0)) !== true) {
+      return (
+        <ButtonPrimary disabled={true}>
+          Enter an amount
+        </ButtonPrimary>
+      );
+    }
+
+    // If account balance is less than inputted amount
+    // This flow should be _before_ approval flow
+    const insufficientFunds = (
+      (balance?.equalTo(JSBI.BigInt(0)) ?? false) ||
+      parsedAmount?.greaterThan(balance)
+    )
+    if (insufficientFunds) {
+      return (
+        <ButtonError error={true} disabled={true}>
+          Insufficient Balance
+        </ButtonError>
+      );
+    }
+
+    // If unapproved, render button approval flow
+    // Approval is only needed for staking
+    if (isStaking) {
+      if (approvalState === ApprovalState.UNKNOWN) {
+        // Disable approve button while approval is being checked
+        return (
+          <ButtonPrimary disabled={true}>
+            Approve
+          </ButtonPrimary>
+        )
+      } 
+      
+      if (approvalState !== ApprovalState.APPROVED) {
+        if (approvalState === ApprovalState.NOT_APPROVED) {
+          return (
+            <ButtonPrimary onClick={handleApproval}>
+              Approve
+            </ButtonPrimary>
+          );
+        }
+
+        if (approvalState === ApprovalState.PENDING) {
+          return (
+            <ButtonPrimary disabled={true} onClick={handleApproval}>
+              <Dots>Approving</Dots>
+            </ButtonPrimary>
+          );
+        }
+      }
+    }
+
+    // Attempt the (un)staking transaction
+    return (
+      <ButtonPrimary
+        disabled={pendingTx}
+        onClick={handleStake}
+      >
+        {isStaking ? 'Stake' : 'Unstake'}
+      </ButtonPrimary>
+    );
+  }
+
+  async function handleStake() {
+    try {
+      setPendingTx(true);
+
+      if (isStaking) {
+        await enter(parsedAmount);
+      } else {
+        await leave(parsedAmount);
+      }
+
+      setInput('');
+    } catch (e) {
+      console.error(`Error ${isStaking ? 'Staking' : 'Unstaking'}: `, e);
+    } finally {
+      setPendingTx(false);
+    }
+  }
+
+  return (
+    <PageWrapper gap="lg" justify="center">
+      <TopSection gap="md">
+        <DataCard>
+          <CardBGImage />
+          <CardNoise />
+          <CardSection>
+            <AutoColumn gap="md">
+              <RowBetween>
+                <TYPE.white fontWeight={600}>Earn more TRI</TYPE.white>
+              </RowBetween>
+              <RowBetween>
+                <TYPE.white fontSize={14}>For every swap on the exchange on every chain, XX% of the swap fees are distributed as TRI
```

Comment:
Can replace text segment with:
"0.05% of every trade on the Trisolaris DEX will be used to buy back TRI and distributed to TRI stakers proportional to your share of the staking pool. When you stake TRI you receive xTRI. Your xTRI is continuously compounding and when you unstake you will receive all of your originally deposited TRI as well as additional earned."